ql/src/java/org/apache/hadoop/hive/ql/io/CombineHiveInputFormat.java

```diff
@@ -370,7 +371,8 @@ private InputSplit[] getCombineSplits(JobConf job, int numSplits,
       PartitionDesc part = HiveFileFormatUtils.getFromPathRecursively(
           pathToPartitionInfo, path, IOPrepareCache.get().allocatePartitionDescMap());
       TableDesc tableDesc = part.getTableDesc();
-      if ((tableDesc != null) && tableDesc.isNonNative()) {
+      boolean useDefaultFileFormat = part.getInputFileFormatClass().equals(tableDesc.getInputFileFormatClass());
```

Comment:
Can `Class.isAssignableFrom` be used here?

Reply:
Done.